packages/core/src/operators/act/act.operator.ts

```diff
@@ -35,17 +37,21 @@ export function act<
     metadata: event.metadata,
   } as ErrorEvent);
 
+  const handleError = (event: InputEvent) => (error: unknown) => pipe(
+    O.fromNullable(errorFn),
+    O.map(fn => fn(error, event)),
+    O.map(res => !isObservable(res) ? of(res) : res),
+    O.getOrElse(() => getDefaultErrorEvent(error)(event)),
+  );
+
   return (source: Observable<InputEvent>): Observable<CallEvent | ErrorEvent> =>
     source.pipe(
-      mergeMap(event => defer(() =>
-        callFn(event).pipe(
-          catchError(error => pipe(
-            O.fromNullable(errorFn),
-            O.map(fn => fn(error, event)),
-            O.map(res => !isObservable(res) ? of(res) : res),
-            O.getOrElse(() => getDefaultErrorEvent(error)(event)),
-          )),
-        ),
-      )),
+      mergeMap(event => defer(() => pipe(
+        E.tryCatch(
```

Comment:
Correct me if I'm wrong, isn't this just catching an error, wrapping it into an `Either`, only to unwrap it again to get to the underlying `ErrorEvent` again? I wonder if this could be simplified without using `Either` here, if the goal is to produce an `Observable` anyways.

Reply:
Well, the idea was to use Either because I don't like explicit `try/catch` statements. But keeping in mind the performance I can change it.